docs/reference/streamfield/blocks.md

```diff
@@ -0,0 +1,462 @@
+- **form_classname** – An HTML `class` attribute to set on the root element of this block as displayed in the editing interface.
+- **min_num** – Minimum number of sub-blocks that the list must have.
+- **max_num** – Maximum number of sub-blocks that the list may have.
+- **collapsed** – When true, all sub-blocks are initially collapsed.
+
+***class* wagtail.blocks.StreamBlock**
+A block consisting of a sequence of sub-blocks of different types, which can be mixed and reordered at will. Used as the overall mechanism of the StreamField itself, but can also be nested or used within other structural block types. Takes a list of `(name, block_definition)` tuples as its first argument:
+```python
+body = StreamField([
+    # ...
+    ('carousel', blocks.StreamBlock(
+        [
+            ('image', ImageChooserBlock()),
+            ('quotation', blocks.StructBlock([
+                ('text', blocks.TextBlock()),
+                ('author', blocks.CharBlock()),
+            ])),
+            ('video', EmbedBlock()),
+        ],
+        icon='cogs'
+    )),
+], use_json_field=True)
+```
+As with StructBlock, the list of sub-blocks can also be provided as a subclass of StreamBlock:
+```python
+class CarouselBlock(blocks.StreamBlock):
+    image = ImageChooserBlock()
+    quotation = blocks.StructBlock([
+        ('text', blocks.TextBlock()),
+        ('author', blocks.CharBlock()),
+    ])
+    video = EmbedBlock()
+
+    class Meta:
+        icon='cogs'
+```
+Since `StreamField` accepts an instance of `StreamBlock` as a parameter, in place of a list of block types, this makes it possible to re-use a common set of block types without repeating definitions:
+```python
+class HomePage(Page):
+    carousel = StreamField(
+        CarouselBlock(max_num=10, block_counts={'video': {'max_num': 2}}),
+        use_json_field=True
+     )
+```
+`StreamBlock` accepts the following additional options as either keyword arguments or `Meta` properties:
+
+**Parameters**
+- **required** - If true (the default), at least one sub-block must be supplied. This is ignored when using the `StreamBlock` as the top-level block of a StreamField; in this case the StreamField’s `blank` property is respected instead.
+- **min_num** – Minimum number of sub-blocks that the stream must have.
+- **max_num** – Maximum number of sub-blocks that the stream may have.
+- **block_counts** – Specifies the minimum and maximum number of each block type, as a dictionary mapping block names to dicts with (optional) `min_num` and `max_num` fields.
+**collapsed** – When true, all sub-blocks are initially collapsed.
+- **form_classname** - An HTML `class` attribute to set on the root element of this block as displayed in the editing interface.
+```python
+body = StreamField([
+    # ...
+    ('event_promotions', blocks.StreamBlock([
+        ('hashtag', blocks.CharBlock()),
+        ('post_date', blocks.DateBlock()),
+    ], form_classname='event-promotions')),
+], use_json_field=True)
+```
```

Comment:
Please check spacing between code blocks and other content (however using eval-rst should avoid a lot of these issues as you can use the existing rst).